pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -517,3 +690,11 @@ func (r *ReconcileUserSignup) updateStatusConditions(userSignup *toolchainv1alph
 	}
 	return r.client.Status().Update(context.TODO(), userSignup)
 }
+
+// validateEmailHash calculates an md5 hash value for the provided userEmail string, and compares it to the provided
+// userEmailHash.  If the values are the same the function returns true, otherwise it will return false
+func validateEmailHash(userEmail, userEmailHash string) bool {
+	md5hash := md5.New()
+	_, _ = md5hash.Write([]byte(userEmail))
```

Comment:
Don't hide the error. Return it.

Reply:
Done.. I've added a new status updater to reflect an error here also.